substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -1172,3 +1179,13 @@ fn copy_blob_to_target_directory(cargo_manifest: &Path, blob_binary: &WasmBinary
 	)
 	.expect("Copies blob binary to `WASM_TARGET_DIRECTORY`.");
 }
+
+// Get jobserver from parent cargo command
+fn get_jobserver() -> &'static Option<jobserver::Client> {
+	static JOBSERVER: OnceLock<Option<jobserver::Client>> = OnceLock::new();
+
+	JOBSERVER.get_or_init(|| {
+		// Unsafe because it deals with raw fds
+		unsafe { jobserver::Client::from_env() }
```

Comment:
Given the documentation, this should be called as early as possible in the process to get the correct fd. This said, we should move this call to a different location. I think at the top of `build_project` should be fine.